server/homebrew.api.js

```diff
@@ -6,13 +6,52 @@ const GoogleActions = require('./googleActions.js');
 const Markdown = require('../shared/naturalcrit/markdown.js');
 const yaml = require('js-yaml');
 const asyncHandler = require('express-async-handler');
+const { nanoid } = require('nanoid');
 
 // const getTopBrews = (cb) => {
 // 	HomebrewModel.find().sort({ views: -1 }).limit(5).exec(function(err, brews) {
 // 		cb(brews);
 // 	});
 // };
 
+const getBrew = (accessType)=>{
+	return async (req, res, next)=>{
+		const { brew } = req;
+
+		if(!brew) {
+			let id = req.params.id, googleId = req.body?.googleId;
+
+			if(id.length > 12) {
+				googleId = id.slice(0, -12);
+				id = id.slice(-12);
+			}
+			let found = await HomebrewModel.get(accessType === 'edit' ? { editId: id } : { shareId: id })
+				.catch((err)=>{
+					if(googleId) {
+						console.warn(`Unable to find document stub for ${accessType}Id ${id}`);
+					} else {
+						console.warn(err);
+					}
+				});
+
+			if(googleId || found?.googleId) {
+				const googleBrew = await GoogleActions.getGoogleBrew(googleId || found?.googleId, id, accessType)
+					.catch((err)=>{
+						console.warn(err);
+					});
+				if(!found && !googleBrew) throw 'Brew not found in database or Google Drive';
+				found = found ? Object.assign(excludeStubProps(found), excludeGoogleProps(googleBrew)) : googleBrew;
```

Comment:
What happens if we find a stub but the associated Google Brew is missing?

Reply:
Good call, I think right now it would try to load the page but would be missing some fields like text, etc...

I need to make sure it accounts for that, in that case do you think it makes sense to have the response result in a 500?

Reply:
For Google Drive we can probably grab whatever error code Google sends and respond with that. In case it's just due to user not logged in/ etc so we can provide an appropriate instruction to the user.